modules/servicetemplate/src/main/java/org/eclipse/xpanse/modules/servicetemplate/ServiceTemplateManage.java

```diff
@@ -129,6 +132,37 @@ private ServiceTemplateEntity getNewServiceTemplateEntity(Ocl ocl) {
         return newTemplate;
     }
 
+
+    private Semver getSemverVersion(String serviceVersion) {
+        try {
+            return new Semver(serviceVersion);
+        } catch (SemverException e) {
+            String errorMsg = String.format("The service version %s is a invalid semver version.",
+                    serviceVersion);
+            throw new IllegalArgumentException(errorMsg);
```

Comment:
we need a new proper exception here - InvalidServiceVersionException

Reply:
Added new exception as suggestion.